cmd/object-handlers.go

```diff
@@ -667,6 +675,14 @@ func (api objectAPIHandlers) PutObjectHandler(w http.ResponseWriter, r *http.Req
 		return
 	}
 
+	// Deny if WORM is enabled
+	if globalWORMEnabled {
+		if _, err = objectAPI.GetObjectInfo(bucket, object); err == nil {
+			writeErrorResponse(w, ErrMethodNotAllowed, r.URL)
+			return
+		}
```

Comment:
@balamurugana the above lines are not needed as the object-layer already does the check before doing the rename